src/main/java/uk/gov/ida/matchingserviceadapter/configuration/EuropeanIdentityConfiguration.java

```diff
@@ -30,10 +37,11 @@ public String getHubConnectorEntityId() {
         return hubConnectorEntityId;
     }
 
-    public String[] getAcceptableHubConnectorEntityIds() {
-        return acceptableHubConnectorEntityIds != null && acceptableHubConnectorEntityIds.length > 0
-            ? acceptableHubConnectorEntityIds
-            : new String[] { hubConnectorEntityId };
+    public List<String> getAcceptableHubConnectorEntityIds(MatchingServiceAdapterEnvironment environment) {
+        Set<String> entityIds = new HashSet<>(DEFAULT_ACCEPTABLE_HUB_CONNECTOR_IDS.getOrDefault(environment, new ArrayList<>()));
+        Optional.ofNullable(acceptableHubConnectorEntityIds).ifPresent(entityIds::addAll);
+        Optional.ofNullable(hubConnectorEntityId).ifPresent(entityIds::add);
```

Comment:
`hubConnectorEntityId ` is always non-null, isn't it?

Reply:
I wouldn't like to assume that in future...

Reply:
It's got the `@NonNull` annotation up top